R/html_dependency.R

```diff
@@ -354,8 +354,18 @@ copyDependencyToDir <- function(dependency, outputDir, mustWork = TRUE) {
   dir.create(target_dir)
   dependency$src$file <- normalizePath(target_dir, "/", TRUE)
 
-  files <- if (dependency$all_files) list.files(dir) else {
-    unlist(dependency[c('script', 'stylesheet', 'attachment')])
+  if (dependency$all_files)
+    files <- list.files(dir)
+  else {
+    file_list <- c(dependency[['script']],
+                   dependency[['stylesheet']],
+                   dependency[['attachment']])
+    if (anyUnnamed(file_list))
+      files <- vapply(file_list,
+                      function(f) if (is.list(f)) f[['src']] else f,
```

Comment:
Does this correctly handle the case of `htmlDependency("foo", "1.0", src="", script=list(list(src="file.js")))`? I think the tests should verify that the files were actually copied in order to verify this

Reply:
Yes, that's worth testing.  I'll add that.